pipeline/translate/splitter.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+"""
+Splits a dataset to chunks. Generates files in format file.00.zst, file.01.zst etc.
+
+Example:
+    python splitter.py \
+        --output_dir=test_data \
+        --compression_cmd=zstd \
+        --num_parts=10 \
+        --output_suffix=.ref \
+        test_data/corpus.en.zst
+"""
+
+import argparse
+import os
+import subprocess
+from contextlib import ExitStack
+
+
+def split_file(mono_path, output_dir, num_parts, compression_cmd, output_suffix=""):
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Initialize the decompression command
+    decompress_cmd = f"{compression_cmd} -dc {mono_path}"
+
+    # Use ExitStack to manage the cleanup of file handlers
+    with ExitStack() as stack:
+        decompressed = stack.enter_context(
```

Comment:
suggestion (non-blocking): consider decompressing to a file on disk, so you only have to do the decompression once. (`zstd -d --rm` will do this - I'm not certain what peak disk usage will look like though.)

(I trust you to know whether it's better to optimize for CPU vs disk here.)

Reply:
It's a good question what would be our preferred best practice. We used to always prioritize having all the data compressed because the datasets are enormous and eat a lot of disk space and I still try to follow this unless working with compressed data is not supported. Now with ephemeral machines and elastic local storage it might make more sense to just decompress everything right away and work with the decompressed data. Maybe when we start training on Berlin cluster having everything compressed can help manage the data easier. We should discuss and unify our strategy.